core_lib_generator/core_lib_template_generator.py

```diff
@@ -0,0 +1,235 @@
+        add_db = input_yes_no('\nDo you want to add another DB connection?', False)
+        db_template[db_name.lower()] = {
+            'env': {
+                f'{db_name.upper()}_USER': db_username,
+                f'{db_name.upper()}_PASSWORD': db_password,
+                f'{db_name.upper()}_PORT': db_port,
+                f'{db_name.upper()}_DB': db_name,
+                f'{db_name.upper()}_HOST': db_host,
+            },
+            'config': {
+                'protocol': DBTypes(db_type).name.lower(),
+                'username': f'${{oc.env:{db_name.upper()}_USER}}',
+                'password': f'${{oc.env:{db_name.upper()}_PASSWORD}}',
+                'host': f'${{oc.env:{db_name.upper()}_HOST}}',
+                'port': f'${{oc.env:{db_name.upper()}_PORT}}',
+                'file': f'${{oc.env:{db_name.upper()}_DB}}',
+            },
+        }
+    return db_template
+
+
+def generate_solr_template() -> dict:
+    solr_protocol = input_str('Enter solr protocol (http/https)', 'https')
+    solr_port = input_int('Enter solr port no.', 8983)
+    solr_host = input_str('Enter solr host', 'localhost')
+    solr_file = input_str('Enter solr file name (solr/core_name)', 'solr/mycore')
+    print('\nInput recorded.')
+    return {
+        'env': {
+            'SOLR_HOST': solr_host,
+            'SOLR_PORT': solr_port,
+        },
+        'config': {
+            'protocol': solr_protocol,
+            'host': f'${{oc.env:SOLR_HOST}}',
+            'port': f'${{oc.env:SOLR_PORT}}',
+            'file': solr_file,
+        },
+    }
+
+
+def generate_cache_template() -> dict:
+    cache_type = input_enum(CacheTypes, 'for cache type')
+
+    if cache_type == 1:
+        memcache_port = input_int('Enter your memcached server port no.', 11211)
+        memcache_host = input_str('Enter your memcached server host', 'localhost')
+        print(f'\nCache type {CacheTypes(cache_type).name} on {memcache_host}:{memcache_port}')
+        return {
+            'env': {
+                'MEMCACHED_HOST': memcache_host,
+                'MEMCACHED_PORT': memcache_port,
+            },
+            'config': {
+                'type': CacheTypes(cache_type).name.lower(),
+                'host': f'${{oc.env:MEMCACHED_HOST}}',
+                'port': f'${{oc.env:MEMCACHED_PORT}}',
+            },
+        }
+    elif cache_type == 2:
+        print(f'\nCache type {CacheTypes(cache_type).name}')
+        return {
+            'config': {
+                'type': CacheTypes(cache_type).name.lower(),
+            }
+        }
+    else:
+        print('\nNo cache set.')
+
+
+def generate_db_entity_template() -> dict:
+    entities = {}
+    add_entity = True
+    while add_entity:
+        is_soft_delete = False
+        is_soft_delete_token = False
+        entity_name = input_str('Enter the name of the entity you\'d like to create: ', 'User')
+        while entity_name in entities:

```

Comment:
no duplicate logic

Reply:
if the user is reentering a name that is already being created it is for that